// @polkadot/util/cjs/is/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "isArray", {
  enumerable: true,
  get: function () {
    return _array.isArray;
  }
});
Object.defineProperty(exports, "isAscii", {
  enumerable: true,
  get: function () {
    return _ascii.isAscii;
  }
});
Object.defineProperty(exports, "isBigInt", {
  enumerable: true,
  get: function () {
    return _bigInt.isBigInt;
  }
});
Object.defineProperty(exports, "isBn", {
  enumerable: true,
  get: function () {
    return _bn.isBn;
  }
});
Object.defineProperty(exports, "isBoolean", {
  enumerable: true,
  get: function () {
    return _boolean.isBoolean;
  }
});
Object.defineProperty(exports, "isBuffer", {
  enumerable: true,
  get: function () {
    return _buffer.isBuffer;
  }
});
Object.defineProperty(exports, "isChildClass", {
  enumerable: true,
  get: function () {
    return _childClass.isChildClass;
  }
});
Object.defineProperty(exports, "isCodec", {
  enumerable: true,
  get: function () {
    return _codec.isCodec;
  }
});
Object.defineProperty(exports, "isCompact", {
  enumerable: true,
  get: function () {
    return _compact.isCompact;
  }
});
Object.defineProperty(exports, "isError", {
  enumerable: true,
  get: function () {
    return _error.isError;
  }
});
Object.defineProperty(exports, "isFunction", {
  enumerable: true,
  get: function () {
    return _function.isFunction;
  }
});
Object.defineProperty(exports, "isHex", {
  enumerable: true,
  get: function () {
    return _hex.isHex;
  }
});
Object.defineProperty(exports, "isInstanceOf", {
  enumerable: true,
  get: function () {
    return _instanceOf.isInstanceOf;
  }
});
Object.defineProperty(exports, "isIp", {
  enumerable: true,
  get: function () {
    return _ip.isIp;
  }
});
Object.defineProperty(exports, "isJsonObject", {
  enumerable: true,
  get: function () {
    return _jsonObject.isJsonObject;
  }
});
Object.defineProperty(exports, "isNull", {
  enumerable: true,
  get: function () {
    return _null.isNull;
  }
});
Object.defineProperty(exports, "isNumber", {
  enumerable: true,
  get: function () {
    return _number.isNumber;
  }
});
Object.defineProperty(exports, "isObject", {
  enumerable: true,
  get: function () {
    return _object.isObject;
  }
});
Object.defineProperty(exports, "isObservable", {
  enumerable: true,
  get: function () {
    return _observable.isObservable;
  }
});
Object.defineProperty(exports, "isPromise", {
  enumerable: true,
  get: function () {
    return _promise.isPromise;
  }
});
Object.defineProperty(exports, "isString", {
  enumerable: true,
  get: function () {
    return _string.isString;
  }
});
Object.defineProperty(exports, "isTestChain", {
  enumerable: true,
  get: function () {
    return _testChain.isTestChain;
  }
});
Object.defineProperty(exports, "isToBigInt", {
  enumerable: true,
  get: function () {
    return _toBigInt.isToBigInt;
  }
});
Object.defineProperty(exports, "isToBn", {
  enumerable: true,
  get: function () {
    return _toBn.isToBn;
  }
});
Object.defineProperty(exports, "isU8a", {
  enumerable: true,
  get: function () {
    return _u8a.isU8a;
  }
});
Object.defineProperty(exports, "isUndefined", {
  enumerable: true,
  get: function () {
    return _undefined.isUndefined;
  }
});
Object.defineProperty(exports, "isUtf8", {
  enumerable: true,
  get: function () {
    return _utf.isUtf8;
  }
});
Object.defineProperty(exports, "isWasm", {
  enumerable: true,
  get: function () {
    return _wasm.isWasm;
  }
});
var _array = require("./array");
var _ascii = require("./ascii");
var _bigInt = require("./bigInt");
var _bn = require("./bn");
var _buffer = require("./buffer");
var _boolean = require("./boolean");
var _childClass = require("./childClass");
var _codec = require("./codec");
var _compact = require("./compact");
var _error = require("./error");
var _function = require("./function");
var _hex = require("./hex");
var _instanceOf = require("./instanceOf");
var _ip = require("./ip");
var _jsonObject = require("./jsonObject");
var _null = require("./null");
var _number = require("./number");
var _object = require("./object");
var _observable = require("./observable");
var _promise = require("./promise");
var _string = require("./string");
var _testChain = require("./testChain");
var _toBigInt = require("./toBigInt");
var _toBn = require("./toBn");
var _u8a = require("./u8a");
var _undefined = require("./undefined");
var _utf = require("./utf8");
var _wasm = require("./wasm");
