// @polkadot/util/cjs/object/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "objectClear", {
  enumerable: true,
  get: function () {
    return _clear.objectClear;
  }
});
Object.defineProperty(exports, "objectCopy", {
  enumerable: true,
  get: function () {
    return _copy.objectCopy;
  }
});
Object.defineProperty(exports, "objectEntries", {
  enumerable: true,
  get: function () {
    return _entries.objectEntries;
  }
});
Object.defineProperty(exports, "objectKeys", {
  enumerable: true,
  get: function () {
    return _keys.objectKeys;
  }
});
Object.defineProperty(exports, "objectProperties", {
  enumerable: true,
  get: function () {
    return _property.objectProperties;
  }
});
Object.defineProperty(exports, "objectProperty", {
  enumerable: true,
  get: function () {
    return _property.objectProperty;
  }
});
Object.defineProperty(exports, "objectSpread", {
  enumerable: true,
  get: function () {
    return _spread.objectSpread;
  }
});
Object.defineProperty(exports, "objectValues", {
  enumerable: true,
  get: function () {
    return _values.objectValues;
  }
});
var _clear = require("./clear");
var _copy = require("./copy");
var _entries = require("./entries");
var _keys = require("./keys");
var _property = require("./property");
var _spread = require("./spread");
var _values = require("./values");
